src/components/components/AddLayerPanel/AddLayerPanel.component.jsx

```diff
@@ -0,0 +1,378 @@
+    } else return 0; // default value
+  };
+
+  const cardMouseEnter = (mixinId) => {
+
+    preEntity.setAttribute('mixin', mixinId);
+    const selectedElement = AFRAME.INSPECTOR.selected?.el;
+    if (selectedElement) {
+      selectedElement.object3D.getWorldPosition(selectedObjPos);
+      // get elevation position Y from attribute of segment element
+      segmentElevationPosY = getSegmentElevationPosY(selectedElement);
+
+      // avoid adding preview element inside the direct ancestor of a-scene: #environment, #reference, ...
+      if (ancestorOfSelectedEl.parentEl.isScene) return;
+
+      preEntity.setAttribute('visible', true);
+      selectedObjPos.setY(segmentElevationPosY);
+      if (inSegment) {
+        preEntity.setAttribute('position', selectedObjPos);
+      } else {
+        preEntity.setAttribute('position', {x: selectedObjPos.x, y: 0.2, z: selectedObjPos.z});
+      }
+    }
+  };
+
+  const cardMouseLeave = (mixinId) => {
+    preEntity.setAttribute('visible', false);
+  };
+
+  const createEntity = (mixinId, parentEl) => {
+    console.log('create entity: ', mixinId);
+    const newEntity = document.createElement('a-entity');
+    newEntity.setAttribute('mixin', mixinId);
+
+    const selectedElement = AFRAME.INSPECTOR.selected?.el;
+
+    // avoid adding new element inside the direct ancestor of a-scene: #environment, #reference, ...
+    if (selectedElement && !ancestorOfSelectedEl.parentEl.isScene) {
+      // append element as a child of the entity with .custom-group class.
+      let customGroupEl = ancestorOfSelectedEl.querySelector('.custom-group');
+      if (!customGroupEl) {
+        customGroupEl = document.createElement('a-entity');
+        // .custom-group entity is a child of segment or .street-parent/.buildings-parent elements
+        ancestorOfSelectedEl.appendChild(customGroupEl);
+        customGroupEl.classList.add('custom-group');
+
+        if (inSegment) {
+          // set position y by elevation level of segment
+          customGroupEl.setAttribute('position', { y: segmentElevationPosY });
+        } else {
+          // if we are creating element not inside segment-parent
+          customGroupEl.setAttribute('position', selectedObjPos);
+        }
+      }
+      customGroupEl.appendChild(newEntity);
+    } else {
+      const streetContainer = document.querySelector('street-container');
+      // apppend element as a child of street-container
+      if (streetContainer) {
+        streetContainer.appendChild(newEntity);
+      } else {
+        AFRAME.scenes[0].appendChild(newEntity);
+      }
+    }
+    Events.emit('entitycreated', newEntity);
+  };
+
+  const cardClick = (card) => {
+    if (card.mixinId) {
+      createEntity(card.mixinId);
+    } else if (card.handlerFunction) {
+      card.handlerFunction();
+    }
+  };
+  return (
+    <div
+      className={classNames(styles.panel, isAddLayerPanelOpen && styles.open)}
```

Comment:
I would have written
```js
classNames(styles.panel, { [styles.open]: isAddLayerPanelOpen })
```
but your syntax works too. It may not be readable if you have more conditions though.